server/src/main/java/org/elasticsearch/indices/cluster/IndicesClusterStateService.java

```diff
@@ -740,6 +743,7 @@ private void createShardWhenLockAvailable(
         long primaryTerm,
         int iteration,
         long delayMillis,
+        AtomicBoolean hasDumpedHotThreads,
```

Comment:
I'm just using `AtomicBoolean` here as a mutable container for a boolean. happy for suggestions of other ways to convey `hasDumpedHotThreads` through the recursive calls. I didn't want to use a plain boolean because I didn't want to reassign a method param or keep a second one.

Reply:
We also have `RunOnce` which feels a better fit?

Reply:
Perfect, thanks!